code/lib/cli/src/automigrate/fixes/new-frameworks.test.ts

```diff
@@ -0,0 +1,223 @@
+/* eslint-disable no-underscore-dangle */
+import path from 'path';
+import { JsPackageManager } from '../../js-package-manager';
+import { newFrameworks } from './new-frameworks';
+
+// eslint-disable-next-line global-require, jest/no-mocks-import
+jest.mock('fs-extra', () => require('../../../../../__mocks__/fs-extra'));
+
+const checkNewFrameworks = async ({ packageJson, main }) => {
+  if (main) {
+    // eslint-disable-next-line global-require
+    require('fs-extra').__setMockFiles({
+      [path.join('.storybook', 'main.js')]: `module.exports = ${JSON.stringify(main)};`,
+    });
+  }
+  const packageManager = {
+    retrievePackageJson: () => ({ dependencies: {}, devDependencies: {}, ...packageJson }),
+  } as JsPackageManager;
+  return newFrameworks.check({ packageManager });
+};
+
+describe('new-frameworks fix', () => {
+  describe('should no-op', () => {
+    it('in sb < 7', async () => {
+      const packageJson = { dependencies: { '@storybook/vue': '^6.2.0' } };
+      await expect(
+        checkNewFrameworks({
+          packageJson,
+          main: {},
+        })
+      ).resolves.toBeFalsy();
+    });
+
+    it('in sb 7 with no main', async () => {
+      const packageJson = { dependencies: { '@storybook/vue': '^7.0.0' } };
+      await expect(
+        checkNewFrameworks({
+          packageJson,
+          main: undefined,
+        })
+      ).resolves.toBeFalsy();
+    });
+
+    it('in sb 7 with no framework field in main', async () => {
```

Comment:
Should we check the addons list for a framework? At once stage that's how people did it. Or are we saying that's out of scope of this automigration and such people need to figure it out on their own?

Reply:
I'd say out of scope because the frameworks we support at the moment don't fall under this scenario. Once we have CRA,Next etc then it would be applicable